demo/src/main/java/com/novoda/storagepathfinder/demo/AssetCloner.kt

```diff
@@ -0,0 +1,27 @@
+package com.novoda.storagepathfinder.demo
+
+import android.content.res.AssetManager
+import android.util.Log
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.concurrent.Executor
+
+internal class AssetCloner(private val assetManager: AssetManager, private val executor: Executor) {
+
+    fun cloneAsset(assetName: String, pathToCloneTo: String) {
+        executor.execute {
+            File(pathToCloneTo).also {
+                it.parentFile?.takeIf { !it.exists() }?.also { it.mkdirs() }
+            }.also { copyAssetToFile(assetName, it) }
+        }
+    }
+
+    private fun copyAssetToFile(assetName: String, outputFile: File) = try {
+        assetManager.open(assetName)
+                .copyTo(FileOutputStream(outputFile, true))
```

Comment:
I actually couldn't believe how easy this section was after writing the java code 😄 so much boilerplate code is gone.

Reply:
Yeah, this is amazing! So is the `copyTo` taking care of closing everything?

Reply:
The only thing I'm not sure about the `copyAssetToFile` function is defining it with the `=` symbol (I don't know how to say it...)
I would have gone with 
```
private fun copyAssetToFile(assetName: String, outputFile: File) {
    try {
    }
}
```

Reply:
Yea it does everything. It's from the standard libs on kotlin https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/index.html

Reply:
Ooooo actually, after reading it again ❌ I need to close these resources.